sycl/doc/extensions/SPIRV/SPV_INTEL_memory_access_aliasing.asciidoc

```diff
@@ -0,0 +1,275 @@
+SPV_INTEL_memory_access_aliasing
+================================
+
+Name Strings
+------------
+
+SPV_INTEL_memory_access_aliasing
+
+Contact
+-------
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm
+
+Contributors
+------------
+
+- Dmitry Sidorov, Intel
+- Alexey Sotkin, Intel
+
+Notice
+------
+
+Copyright (c) 2021 Intel Corporation.  All rights reserved.
+
+Status
+------
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a
+feature for review and community feedback. When the feature matures, this
+specification may be released as a formal extension.
+
+Because the interfaces defined by this specification are not final and are
+subject to change they are not intended to be used by shipping software
+products.
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2021-02-15
+| Revision           | A
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification, Version 1.5
+Revision 5.
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension adds *AliasDomainDeclINTEL*, *AliasScopeINTEL* and
+*AliasScopeListDeclINTEL* instructions that serve a purpose to
+specify aliasing scopes within which annotated memory accesses don't alias.
+Results of these instructios are used by *AliasScopeINTEL* and *NoAliasINTEL*
+decorations and Memory Operands mask to annotate memory access instructions.
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following *OpExtension* must
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_memory_access_aliasing"
+----
+
+New Capabilities
+----------------
+This extension introduces a new capability:
+
+----
+MemoryAccessAliasingINTEL
+----
+
+New instructions
+----------------
+
+Instructions added under the *MemoryAccessAliasingINTEL* capability:
+
+----
+AliasDomainDeclINTEL
+AliasScopeDeclINTEL
+AliasScopeListDeclINTEL
+----
+
+New Decorations
+---------------
+Decorations added under the *MemoryAccessAliasingINTEL* capability:
+
+----
+AliasScopeINTEL
+NoAliasINTEL
+----
+
+Token Number Assignments
+------------------------
+[width="45%",cols="30,15"]
+|=================================
+| MemoryAccessAliasingINTEL | 5910
+| AliasDomainDeclINTEL      | 5911
+| AliasScopeDeclINTEL       | 5912
+| AliasScopeListDeclINTEL   | 5913
+| AliasScopeINTEL           | 5914
+| NoAliasINTEL              | 5915
+|=================================
+
+Modifications to the SPIR-V Specification, Version 1.5
+------------------------------------------------------
+
+Modifications 2.4 Logical Layout of a Module
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Insert an extra item before "8. All annotation instructions"
+
+8. Optional _AliasDomainDeclINTEL_, _AliasScopeDeclINTEL_ and _AliasScopeDeclINTEL_
+instructions.
```

Comment:
If I understand correctly, all annotation instructions have no result ID, instead the have a target ID to where the instruction is applied. In the same time, aliasing instructions have the result ID, which is used by decoration instruction and memory access mask. In general, I don't appose this suggestion, because aliasing instructions by meaning are some sort of annotations, but it might be confusing for spec readers to see these instruction among result-less annotation instructions.

Reply:
>  In general, I don't appose this suggestion, because aliasing instructions by meaning are some sort of annotations, but it might be confusing for spec readers to see these instruction among result-less annotation instructions.

Ok, fair enough. I don't have any strong preferences, current description looks fine to me as well